haris2055/SOEN390
# Notifications Features

This pull request includes the implementation of the following user stories:

#126 [US-20] Connection Invitation Notification
To test this feature:
- Received Connection Request
  - go to login page
  - sign in to a valid account
  - have another user send you a Connection request
  - go to the Notifications page, and there should be a notification which says that you've received a Connection request from that use
-  Sent Connection Request
    - go to login page
    - sign in to a valid account
    - send another user a Connection request
    - have the other user accept the Connection request
    - go to the Notification page, and there should be a notification which says that the other user has accepted your Connection request  

#133 [US-27] Legal "Backdoor" Notification
To test this feature:
- go to register page
- register for a new account
- accept the confirmation email
- log in to your new account
- go to the Notification page, and there should be a notification describing the legal backdoor policy

#171 [US-42-43-V2] Job Alerts
To test this feature:
- Saving Job Alerts
  - go to login page
  - sign in to a valid account
  - navigate to the Job tab of the search results
  - press the Filter button to open the Search Filter form
  - fill out some values into the Fitler Form that you want your Job Alert to be based on
  - press the Save Search button to save the your Search Filter as a Job Alert
  - create a Job that matches the parameters you saved
  - you should receive a notification that a Job has been posted that matches one of your saved Job Alerts

- Managing Job Alerts:
  - in the header, click the Profile drop down menu and click Settings
  - if you have saved any searches as Job Alerts, they should appear below the Forgot Password form
  - try deleting some of your existing Job Alerts

#146 [US-51] Notifications Page
To test this feature:
- try receiving Connection requests and have other users accept your Connection requests
  - try interacting with the Connection request (accept/reject) via the dropdown menu on the notification card
- try receiving Recommendations from other users
  - try viewing the profile of the user who recommended you via the dropdown menu on the notification card
- try saving Job Alerts and have Jobs posted that match any of your saved Job Alerts
  - try viewing the job via the dropdown menu on the notification card
- unread Notification count in Header should update automatically
- when on the Notifications page, Notifications should update automatically 

If merged, this pull request closes #126, closes #133, closes #171, and closes #146.
